backend/app/auth_handlers.go

```diff
@@ -0,0 +1,246 @@
+package app
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/codescalersinternships/Flyspray/internal"
+	"github.com/codescalersinternships/Flyspray/models"
+	"gorm.io/gorm"
+
+	"github.com/gin-gonic/gin"
+)
+
+func (a *App) Signup(ctx *gin.Context) (interface{}, Response) {
+	var user models.User
+
+	err := json.NewDecoder(ctx.Request.Body).Decode(&user)
+
+	if err != nil {
+		return nil, BadRequest(err)
+	}
+	err = user.Validate()
+	if err != nil {
+		return nil, BadRequest(fmt.Errorf("validation error: %v", err))
+	}
+
+	hash, err := internal.HashPassword(user.Password)
+
+	if err != nil {
+		return nil, BadRequest(err)
+	}
+
+	user.Password = hash
+
+	user.VerificationCode = a.client.GenerateVerificationCode()
+
+	user, err = a.client.CreateUser(user)
+
+	if err != nil {
+		if err.Error() == "UNIQUE constraint failed: users.email" {
+			return nil, Conflict(errors.New("email already exists"))
+		}
+		return nil, BadRequest(err)
+	}
+
+	err = internal.SendEmail(user.Email, user.VerificationCode)
+	if err != nil {
+
+		return nil, InternalServerError(err)
+	}
+
+	message := "A verification code has been sent to your email."
+
+	return ResponseMsg{Message: message}, Created()
+}
+
+func (a *App) Verify(ctx *gin.Context) (interface{}, Response) {
+
+	var requestBody struct {
+		VerificationCode string `json:"verification_code"`
+	}
+
+	err := json.NewDecoder(ctx.Request.Body).Decode(&requestBody)
+
+	if err != nil {
+		return nil, BadRequest(err)
+	}
+
+	result := a.client.Client.Model(&models.User{}).Where("Verification_code = ?", requestBody.VerificationCode).Update("Verified", true)
+
+	if result.RowsAffected != 1 {
+		return nil, BadRequest(errors.New("wrong verification code"))
+	}
+
+	msg := "your account has been verified successfully"
+	return ResponseMsg{Message: msg}, Ok()
+
+}
+
+func (a *App) SignIn(ctx *gin.Context) (interface{}, Response) {
+
+	var requestBody models.User
+	err := json.NewDecoder(ctx.Request.Body).Decode(&requestBody)
+	if err != nil {
+
+		return nil, BadRequest(err)
+	}
+
+	var user models.User
+	result := a.client.Client.First(&user, "Email = ?", requestBody.Email)
+
+	if result.Error != nil {
+		if result.Error == gorm.ErrRecordNotFound {
+			return nil, NotFound(errors.New("invalid email or password"))
+		}
+		return nil, InternalServerError(result.Error)
+	}
+
+	isPasswordMatches := internal.CheckPasswordHash(requestBody.Password, user.Password)
+
+	if !isPasswordMatches {
+
+		return nil, NotFound(errors.New("invalid email or password"))
+	}
+
+	if !user.Verified {
+		return nil, Forbidden(errors.New("account not verified"))
```

Comment:
what about `account is not verified yet, please check the verification email in your inbox`?